### ISSUE TITLE
vaadin-crud inside vaadin-custom-field causes instant validation

### ISSUE THREAD

**Issue by FelixGaebler** [vaadin/flow-components#5364]
### Description

I added a `Crud` like in the docs to a `CustomField` as a child. 
However when clicking on the cruds 'New item'-button the form validates instantly.

Also see: https://discord.com/channels/732335336448852018/1140955551002525737

### Expected outcome

The `asRequired()` validations should fire when clicking on the 'Save'-button, but not when opening the form with (always) empty fields.

### Minimal reproducible example

```java
public class CrudField<E extends AbstractEntity> extends CustomField<List<E>> {

    private final List<E> value;

    private final Crud<E> crud;

    private final CustomListDataProvider<E> dataProvider;

    public CrudField(Class<E> clazz, CrudEditor<E> editor) {
        this.value = new ArrayList<>();

        this.dataProvider = new CustomListDataProvider<>(this.value);

        this.crud = new Crud<>(clazz, editor);
        this.crud.setEditOnClick(true);
        this.crud.setDataProvider(this.dataProvider);

        this.crud.addSaveListener(e -> {
            this.value.remove(e.getItem());
            this.value.add(e.getItem());
            updateValue();
        });

        this.crud.addDeleteListener(e -> {
            this.value.remove(e.getItem());
            updateValue();
        });

        setHeightFull();
        addClassName(LumoUtility.Display.BLOCK);
        add(this.crud);
    }

    @Override
    protected List<E> generateModelValue() {
        return new ArrayList<>(this.value);
    }

    @Override
    protected void setPresentationValue(List<E> items) {
        this.value.clear();
        this.value.addAll(items);
    }

}
```

### Steps to reproduce

1. Add CrudField to a view
2. Click on 'New item'-button
3. Watch all fields already validated

### Environment

Vaadin version(s): 24.1.2
OS: Ubuntu 22.04.3 LTS

### Browsers

Issue is not browser related

**Comment by yuriy-fix** [comment]
At the moment `Crud` validation within `custom-field` is not a supported use-case, because `Crud` is not an input field itself. The list of components for which custom-field is designed to be used with are available in the documentation: https://vaadin.com/docs/latest/components/custom-field#basic-usage. 
A proper documentation should be added on how to bind the crud, so considering this issue as a documentation issue.

**Comment by FelixGaebler** [comment]
> At the moment `Crud` validation within `custom-field` is not a supported use-case, because `Crud` is not an input field itself. The list of components for which custom-field is designed to be used with are available in the documentation: https://vaadin.com/docs/latest/components/custom-field#basic-usage. A proper documentation should be added on how to bind the crud, so considering this issue as a documentation issue.

The documentation indicates that the CrudField has been optimized for specified simple field components. There is actually no talk of unsupported here.

One component (Crud) does not behave as expected and the problem does not depend on the programming (without CustomField it works) so according to specification it is a bug.

We also don't want to include workarounds in the documentation, so why should this become a documentation issue?

**Comment by rolfsmeds** [comment]
I don't think you can expect CustomField to work perfectly with _anything_ you put in it. Think about it: you could put, say, an AppLayout inside it and then complain that it doesn't render correctly.

CRUD is intended to be used as the main contents of a view. It is in no way intended to be wrapped inside a field component.

The fact that the documentation doesn't explicitly mention all the unsupported ways CRUD and/or CustomField could be used doesn't mean anything and everything is officially supported unless explicitly mentioned.

We can try to find a more clear way of expressing supported use cases, however, which makes this a documentation issue.

**Comment by FelixGaebler** [comment]
That's what the Vaadin way seems to me to do things. We pay a lot of money for the framework and every bug is declared a documentation issue in no time. If you don't want to solve the problem, you just change the description.

Either way, I understand the argument, even if the example was given about UI and not UX.

But again, let's look at exactly the flaw of the issue and not the architecture as such.

What makes the form in CRUD validate instant? Regardless of the CustomField as wrapper, this should not happen..

**Comment by rolfsmeds** [comment]
We simply cannot predict every possible way someone might decide to use a component, and enumerate every single one of them in the documentation saying whether it's intended/supported or not. We also cannot "fix" every unintended combination that people may want to use, especially if it doesn't strike us as a particularly good idea, as that would take a disproportionate amount of capacity away from making sure that things work correctly when used as intended. So the best we can do is improve the documentation whenever we discover misunderstandings like this.

But we'll have a look to see if we can identify the mechanism is resulting in this instant validation when used in a CustomField like this, to see if we can provide a workaround that avoids it.

**Comment by FelixGaebler** [comment]
This might be related to #5399 and #5319. We will see.
Since I do not think I will be helped here we can also like to close the issue so that there are not more open :)

**Comment by vursen** [comment]
> This might be related to https://github.com/vaadin/flow-components/issues/5399 and https://github.com/vaadin/flow-components/issues/5319. We will see.

I tried to reproduce the issue to see if it might potentially get resolved by this [upcoming validation refactor](https://github.com/orgs/vaadin/discussions/4434#discussioncomment-7067836). However, the issue wasn't reproducible. All the fields appear as "not validated" when I open the "New Item" form: there is no red background, no invalid state, nothing.

Here is a simplified version that I tested:

```java
@Route("crud")
public class CrudView extends Div {
    public class Person {
        private Integer id;
        private String firstName;
        private String lastName;

        public Person(Integer id, String firstName, String lastName) {
            this.id = id;
            this.firstName = firstName;
            this.lastName = lastName;
        }

        public String getFirstName() {
            return firstName;
        }

        public void setFirstName(String firstName) {
            this.firstName = firstName;
        }

        public String getLastName() {
            return lastName;
        }

        public void setLastName(String lastName) {
            this.lastName = lastName;
        }

        public Integer getId() {
            return id;
        }

        @Override
        public int hashCode() {
            return id;
        }

        @Override
        public boolean equals(Object obj) {
            if (this == obj) {
                return true;
            }
            if (!(obj instanceof Person)) {
                return false;
            }
            return false;
        }
    }

    public class PersonDataProvider extends AbstractBackEndDataProvider<Person, CrudFilter> {

        // A real app should hook up something like JPA
        final List<Person> DATABASE = List.of(new Person(0, "Sergey", "Vinogradov"));

        @Override
        protected Stream<Person> fetchFromBackEnd(Query<Person, CrudFilter> query) {
            int offset = query.getOffset();
            int limit = query.getLimit();

            Stream<Person> stream = DATABASE.stream();
            return stream.skip(offset).limit(limit);
        }

        @Override
        protected int sizeInBackEnd(Query<Person, CrudFilter> query) {
            long count = fetchFromBackEnd(query).count();
            return (int) count;
        }
    }

    public class CrudField extends CustomField<List<Person>> {

        private final List<Person> value;

        private final Crud<Person> crud;

        public CrudField() {
            this.value = new ArrayList<>();

            this.crud = new Crud<>(Person.class, createEditor());
            this.crud.setEditOnClick(true);
            this.crud.setDataProvider(new PersonDataProvider());

            this.crud.addSaveListener(e -> {
                this.value.remove(e.getItem());
                this.value.add(e.getItem());
                updateValue();
            });

            this.crud.addDeleteListener(e -> {
                this.value.remove(e.getItem());
                updateValue();
            });

            setHeightFull();
            addClassName(LumoUtility.Display.BLOCK);
            add(this.crud);
        }

        private CrudEditor<Person> createEditor() {
            TextField firstName = new TextField("First name");
            TextField lastName = new TextField("Last name");
            FormLayout form = new FormLayout(firstName, lastName);

            Binder<Person> binder = new Binder<>(Person.class);
            binder.forField(firstName).asRequired().bind(Person::getFirstName, Person::setFirstName);
            binder.forField(lastName).asRequired().bind(Person::getLastName, Person::setLastName);

            return new BinderCrudEditor<>(binder, form);
        }

        @Override
        protected List<Person> generateModelValue() {
            return new ArrayList<>(this.value);
        }

        @Override
        protected void setPresentationValue(List<Person> items) {
            this.value.clear();
            this.value.addAll(items);
        }
    }

    public CrudView() {
        add(new CrudField());
    }
}
```


https://github.com/vaadin/flow-components/assets/5039436/2462e2ca-847f-44f5-8b81-fbeba86d920d

**Comment by felix-office** [comment]
@vursen 
I put exactly your code into a clean vaadin 24.1.10 environment.
First of all the code you provided throws a bug when clicking "new item" because it seems like the bean cannot be initiated. I moved the classes from subclasses to first level classes and created a no args constructor so the bean initialization works.
Still, just using your provided code, I can reproduce the issue:
![Peek 2023-09-30 13-08](https://github.com/vaadin/flow-components/assets/138027957/b905a7ea-64e4-4805-ac9b-6598aeca21f7)


You can check out the project and code here: https://github.com/felix-office/vaadin-bugs/tree/master/src/main/java/de/dataport/bugs/views/report0rep

**Comment by vursen** [comment]
> First of all, the code you provided throws a bug when clicking "new item" because it seems like the bean cannot be initiated.

Oh, indeed, thank you for pointing this out. I checked the corrected example and I can now reproduce the issue in 24.1. However, it's not reproducible in 24.2. Further investigation showed that the issue got fixed by https://github.com/vaadin/web-components/pull/6232.

**Comment by felix-office** [comment]
@vursen Thanks for checking! I can confirm that. Feel free to close the issue or leave it open until 24.2 release. I don't know what your workflow looks like.
